### PR TITLE
Migrate tests a-c to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/test/analytics_prompt_test.dart [hunk 1]
```diff
@@ -17,13 +17,13 @@ import 'package:flutter_test/flutter_test.dart';
...
 
-  bool didCallEnableAnalytics;
+  bool? didCallEnableAnalytics;
 
   Widget _wrapWithAnalytics(
     Widget child, {
-    AnalyticsController controller,
+    AnalyticsController? controller,
   }) {
     _controller ??= controller;
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/analytics_prompt_test.dart:28]: This is a bug. If the controller was specified, we should use it. Replace with:
```
if (controller != null) {
  _controller = controller;
}
```

File: packages/devtools_app/test/analytics_prompt_test.dart [hunk 2]
```diff
@@ -17,13 +17,13 @@ import 'package:flutter_test/flutter_test.dart';
 const windowSize = Size(2000.0, 1000.0);
 
 void main() {
-  AnalyticsController _controller;
+  AnalyticsController? _controller;
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/analytics_prompt_test.dart:20]: late

File: packages/devtools_app/test/app_size_screen_test.dart [hunk 3]
```diff
@@ -48,15 +48,15 @@ void main() {
     data: json.decode(newV8),
   );
 
-  AppSizeScreen screen;
-  AppSizeTestController appSizeController;
+  late AppSizeScreen screen;
+  late AppSizeTestController appSizeController;
   FakeServiceManager fakeServiceManager;
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/app_size_screen_test.dart:53]: shouldn't this be late too?

File: packages/devtools_app/test/app_size_screen_test.dart [hunk 4]
```diff
@@ -84,8 +84,8 @@ void main() {
       appSizeController = AppSizeTestController();
       fakeServiceManager = FakeServiceManager();
       setGlobal(ServiceConnectionManager, fakeServiceManager);
-      when(fakeServiceManager.errorBadgeManager.errorCountNotifier(any))
-          .thenReturn(ValueNotifier<int>(0));
+      // when(fakeServiceManager.errorBadgeManager.errorCountNotifier(any))
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/app_size_screen_test.dart:87]: why is this commented out

File: packages/devtools_app/test/app_size_screen_test.dart [hunk 5]
```diff
@@ -343,11 +343,11 @@ void main() {
 
     Future<void> pumpAppSizeScreenWithContext(
       WidgetTester tester, {
-      AppSizeTestController controller,
+      AppSizeTestController? controller,
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/app_size_screen_test.dart:346]: make this required

File: packages/devtools_app/test/app_size_screen_test.dart [hunk 6]
```diff
@@ -451,9 +451,9 @@ class AppSizeTestController extends AppSizeController {
 
   @override
   void loadDiffTreeFromJsonFiles({
-    @required DevToolsJsonFile oldFile,
-    @required DevToolsJsonFile newFile,
-    @required void Function(String error) onError,
+    required DevToolsJsonFile? oldFile,
+    required DevToolsJsonFile? newFile,
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/app_size_screen_test.dart:455]: make non nullable on the test controller and the super class

File: packages/devtools_app/test/code_size_attribution_test.dart [hunk 7]
```diff
@@ -162,7 +162,8 @@ void main() {
       expect(root.isExpanded, isTrue);
       expect(root.children.length, equals(18));
 
-      for (DominatorTreeNode child in root.children) {
+      for (DominatorTreeNode child
+          in root.children as Iterable<DominatorTreeNode>) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/code_size_attribution_test.dart:166]: use cast instead

File: packages/devtools_app/test/auto_dispose_mixin_test.dart [hunk 8]
```diff
@@ -142,7 +140,8 @@ void main() {
     final controller = StreamController();
     await tester.pumpWidget(AutoDisposedWidget(controller.stream, key: key));
 
-    final _AutoDisposedWidgetState state = key.currentState;
+    final _AutoDisposedWidgetState state =
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/auto_dispose_mixin_test.dart:143]: remove LHS type

File: packages/devtools_app/test/code_size_attribution_test.dart [hunk 9]
```diff
@@ -185,7 +186,8 @@ void main() {
       expect(root.children.length, equals(18));
 
       // Only the selected node should be expanded.
-      for (DominatorTreeNode child in root.children) {
+      for (DominatorTreeNode child
+          in root.children as Iterable<DominatorTreeNode>) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/code_size_attribution_test.dart:190]: cast

File: packages/devtools_app/test/cpu_profiler_controller_test.dart [hunk 10]
```diff
@@ -16,7 +16,7 @@ import 'test_data/cpu_profile_test_data.dart';
 
 void main() {
   group('CpuProfileController', () {
-    CpuProfilerController controller;
+    late CpuProfilerController controller;
     FakeServiceManager fakeServiceManager;
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/cpu_profiler_controller_test.dart:20]: late

File: packages/devtools_app/test/cpu_profiler_test.dart [hunk 11]
```diff
@@ -28,8 +28,8 @@ import 'test_data/cpu_profile_test_data.dart';
 
 void main() {
   CpuProfiler cpuProfiler;
-  CpuProfileData cpuProfileData;
-  CpuProfilerController controller;
+  late CpuProfileData cpuProfileData;
+  late CpuProfilerController controller;
   ServiceConnectionManager fakeServiceManager;
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/cpu_profiler_test.dart:33]: shouldn't these all be late?

File: packages/devtools_app/test/fixtures/flutter_error_app/lib/main.dart [hunk 12]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// @dart=2.9
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/fixtures/flutter_error_app/lib/main.dart:5]: why are we "unmigrating" these files

File: packages/devtools_app/test/analytics_prompt_test.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/test/analytics_prompt_test.dart:28]: done

File: packages/devtools_app/test/analytics_prompt_test.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/test/analytics_prompt_test.dart:20]: done

File: packages/devtools_app/test/app_size_screen_test.dart (hunk 4 above)
Comment (polina-c) [reply packages/devtools_app/test/app_size_screen_test.dart:87]: Forgot to uncomment. Excluded this file from PR to handle it separately.

File: packages/devtools_app/test/app_size_screen_test.dart (hunk 5 above)
Comment (polina-c) [reply packages/devtools_app/test/app_size_screen_test.dart]: postponed the file

File: packages/devtools_app/test/app_size_screen_test.dart (hunk 6 above)
Comment (polina-c) [reply packages/devtools_app/test/app_size_screen_test.dart]: postponed the file

File: packages/devtools_app/test/code_size_attribution_test.dart (hunk 7 above)
Comment (polina-c) [reply packages/devtools_app/test/code_size_attribution_test.dart:166]: done

File: packages/devtools_app/test/auto_dispose_mixin_test.dart (hunk 8 above)
Comment (polina-c) [reply packages/devtools_app/test/auto_dispose_mixin_test.dart:143]: done

File: packages/devtools_app/test/code_size_attribution_test.dart (hunk 9 above)
Comment (polina-c) [reply packages/devtools_app/test/code_size_attribution_test.dart:190]: done

File: packages/devtools_app/test/cpu_profiler_controller_test.dart (hunk 10 above)
Comment (polina-c) [reply packages/devtools_app/test/cpu_profiler_controller_test.dart:20]: done

File: packages/devtools_app/test/cpu_profiler_test.dart (hunk 11 above)
Comment (polina-c) [reply packages/devtools_app/test/cpu_profiler_test.dart:33]: done

File: packages/devtools_app/test/fixtures/flutter_error_app/lib/main.dart (hunk 12 above)
Comment (polina-c) [reply packages/devtools_app/test/fixtures/flutter_error_app/lib/main.dart]: oh, seems to be result of merge. cleared

File: packages/devtools_app/test/cpu_profiler_controller_test.dart (hunk 10 above)
Comment (kenzieschmoll) [reply packages/devtools_app/test/cpu_profiler_controller_test.dart:20]: why doesn't `fakeServiceManager` have to be late here? I'm surprised this isn't giving an analyzer warning

**Review by kenzieschmoll — APPROVED** [review]
one remaining comment then lgtm

**Pull request merged**